src/routes/roleAndPermission.routes.ts

```diff
@@ -0,0 +1,48 @@
+import { Router } from "express";
+import {
+  addPermissions,
+  removePermissions,
+  setRoleAndPermission,
+  setRoleToUser,
+} from "../controllers/RoleAndPermission.controller";
+import { permisionCheck } from "../middleware/permissionCheck/permission.middleware";
+import { ValidateJoi } from "../middleware/validation";
+import { RoleAndPermissionsSchema } from "../middleware/validation/RoleAndPermission.schema.middleware";
+import { protectRoute } from "../services/protectRoutes.service";
+
+const SET_PERMISSION = process.env.SET_PERMISSION as string;
+const RM_PERMISSION = process.env.RM_PERMISSION as string;
+const ADD_PERMISSION = process.env.ADD_PERMISSION as string;
+const SET_ROLE = process.env.SET_ROLE as string;
+
+const setRoleAndPermissionRouter = Router();
+
+setRoleAndPermissionRouter.patch(
+  "/update/:id",
+  ValidateJoi(RoleAndPermissionsSchema.roleTobeSet),
+  [protectRoute, permisionCheck(SET_ROLE)],
```

Comment:
Validation should come after authentication, right?

I think the validation middleware should come as the last check in the process.

Reply:
I think the same applies to other routes.